lib/miq_expression.rb

```diff
@@ -1606,8 +1616,10 @@ def self.ruby_for_date_compare(col_ruby, col_type, tz, op1, val1, op2 = nil, val
 
   def to_arel(exp, tz)
     operator = exp.keys.first
-
-    field = Field.parse(exp[operator]["field"]) if exp[operator].kind_of?(Hash) && exp[operator]["field"]
+    if exp[operator].kind_of?(Hash)
+      field = Field.parse(exp[operator]["field"]) if exp[operator]["field"]
+      exp[operator]["value"] = Field.parse(exp[operator]["value"]).arel_attribute if exp[operator]["value"] && Field.valid_field?(exp[operator]["value"])
```

Comment:
It is probably impolite to change the data that's passed to this method even if it doesn't break anything. Perhaps consider duping or assigning to a temp?